File: SQLiteDB/app/src/main/java/org/texchtown/sqlitedb/MainActivity.java
package org.texchtown.sqlitedb;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";
    //var
    MainFragment mainFragment;
    FragmentManager manager;
    DBHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainFragment = new MainFragment();
        manager = getSupportFragmentManager();
        manager.beginTransaction().replace(R.id.container, mainFragment).commit();

    }

    @Override
    protected void onDestroy() {
        dbHelper = new DBHelper(this);
        dbHelper.close();
        super.onDestroy();
    }
}

File: Starbucks/app/src/main/java/org/texchtown/starbucks/MainActivity.kt
package org.texchtown.starbucks

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.widget.NestedScrollView
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton

class MainActivity : AppCompatActivity() {
    private lateinit var extendedFloatingActionButton: ExtendedFloatingActionButton
    private lateinit var nestedScrollView: NestedScrollView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        extendedFloatingActionButton = findViewById(R.id.ext_fab)
        nestedScrollView = findViewById(R.id.nested_sv)

        nestedScrollView.setOnScrollChangeListener { v, scrollX, scrollY, oldScrollX, oldScrollY ->
            if(scrollY >0) {
                extendedFloatingActionButton.shrink()
            }
            else {
                extendedFloatingActionButton.extend()
            }
        }

        val window = this.window
        window.statusBarColor = ContextCompat.getColor(this, R.color.toolbar_color)
    }
}